### PR TITLE
Accumulate a batch of Skia objects that will be destructed later on the IO thread

### PR DESCRIPTION
**Pull request by jason-simmons**
See https://github.com/dart-lang/sdk/issues/29971

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde @rmacnak-google

File: lib/ui/painting/utils.cc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// found in the LICENSE file.
+
+#include "flutter/common/threads.h"
+#include "flutter/lib/ui/painting/utils.h"
+
+namespace blink {
+
+namespace {
+
+constexpr ftl::TimeDelta kDrainDelay = ftl::TimeDelta::FromMilliseconds(250);
```
Comment (chinmaygarde) [inline lib/ui/painting/utils.cc]: Rather uncomfortable with ~15 frames worth of delay in collecting items. Can this be significantly lower? Say ~20ms?

File: lib/ui/painting/utils.h [hunk 2]
```diff
@@ -2,15 +2,35 @@
...
+ public:
+  static SkiaUnrefQueue& Get();
+
+  void Unref(SkRefCnt* object);
+
+ private:
+  SkiaUnrefQueue();
+  void Drain();
+
+  static SkiaUnrefQueue instance_;
```
Comment (chinmaygarde) [inline lib/ui/painting/utils.h]: We avoid static initialization. Maybe add a `SkiaUnrefQuue::Set` here that initializes the object once during engine initialization and add an assertion in the `Get`.

**Comment by rmacnak-google** [comment]
LGTM - Looks effective on iOS.

**Pull request merged**